### PR TITLE
[C++] Fix typo in README.md

### PR DESCRIPTION
**Pull request by rsabhi**
Fixed minor typos in README.md file

### CODE REVIEW

File: cpp/README.md [hunk 1]
```diff
@@ -110,8 +110,8 @@ ASAN, and `ARROW_USE_ASAN` is mutually-exclusive with the valgrind option
 ### Building/Running fuzzers
 
 Fuzzers can help finding unhandled exceptions and problems with untrusted input that
-may lead to crashes, security issues and undefined behavior. They do this by
-generating random input data and observing the behavior of the executed code. To build
```
Comment (robertnishihara) [inline cpp/README.md]: "behavior" appears to be the preferred spelling in the US http://grammarist.com/spelling/behavior-behaviour/

**Comment by wesm** [comment]
Changed the PR title to be more specific and identify the component modified

File: cpp/README.md (hunk 1 above)
Comment (wesm) [reply cpp/README.md]: Agree, please revert

**Review by wesm — APPROVED** [review]
+1

**Pull request merged**